### ISSUE TITLE
IGraphClient class is not Mockable

### ISSUE THREAD

**Issue by devanshgoenka97** [evo-terren/Gremlin.Net.CosmosDb#65]
The `GraphResult `class is not available publicly and hence  `IGraphClient` is not Mockable, thus disabling the case to create Unit Tests for the same

**Comment by BenjaBobs** [comment]
Meh, the whole thing should really just be interfaced. I don't know why it wasn't in the first place (probably laziness). I don't really have time to work on this, but I guess it's always good to have more tests. I'll review a pr if you would care to interface the thing.

**Comment by amigup** [comment]
@devanshgoenka97 and @BenjaBobs - The IGraphClient reruns the concrete object of `GraphResult` ([link](https://github.com/evo-terren/Gremlin.Net.CosmosDb/blob/master/src/Gremlin.Net.CosmosDb/IGraphClient.cs#L23)) so not sure how interfacing the `GraphResult` would work here. The `GraphResult` constructor should be made public so that it can be mocked by inheriting it to MockClass.

**Comment by BenjaBobs** [comment]
The code's been merged. I do have a question though, why would interfacing it not make it mockable? Both [Moq](https://github.com/Moq/moq4/wiki/Quickstart) and [NSubstitute](https://nsubstitute.github.io/) support creating mocks for interfaces. Since `IGraphClient` is an interface, it could be mocked to return a mocked `IGraphResult` which can then be setup to do whatever the test needs. Am I missing something?

**Comment by devanshgoenka97** [comment]
The issue isn't to mock `IGraphClient`. It is more about creating an object of `GraphResult`.

Also could you build the project so that I can use the latest version on Nuget?

**Comment by BenjaBobs** [comment]
I have zero experience, and right now I don't have time. I might have time to look at it this weekend. We really should get some ci/cd running. Do any of you have any experience with Azure Pipelines or other similar projects?

**Comment by devanshgoenka97** [comment]
Yes, @amigup and myself have experience with Azure Pipleline, and CI/CD in general

**Comment by BenjaBobs** [comment]
Nice! So where does one start with getting that setup? I heard Azure Pipelines is free for open source so that seems like the obvious choice to me.

**Comment by devanshgoenka97** [comment]
You can setup a new Organization and start a new Project [here](https://github.com/marketplace/azure-pipelines)

**Comment by BenjaBobs** [comment]
Even though it's free, it still requires me to have my credit card hooked up, which I'm not too fond of since this is not even my repo. @evo-terren Any chance you can set this up?

**Comment by BenjaBobs** [comment]
Unfortunately, my company is also not very invested in this right now.

**Comment by amigup** [comment]
@BenjaBobs and @evo-terren - If you can grant me access on this repository then I can try creating the build definition using my account. Please grant access to github user - amigup and microsoft live account - amigup@hotmail.com.
